web/src/state_data.ts

```diff
@@ -21,6 +21,32 @@ export const narrow_term_schema = z.object({
 export type NarrowTerm = z.output<typeof narrow_term_schema>;
 // Sync this with zerver.lib.events.do_events_register.
 
+const hotspot_location_schema = z.object({
+    element: z.string(),
+    offset_x: z.number(),
+    offset_y: z.number(),
+    popover: z.optional(z.string()),
+});
+
+const hotspot_schema = z.object({
+    delay: z.number(),
+    description: z.string(),
+    has_trigger: z.boolean(),
+    location: z.optional(hotspot_location_schema),
+    name: z.string(),
+    title: z.string(),
+    type: z.literal("hotspot"),
+});
+
+const one_time_notice_schema = z.object({
+    name: z.string(),
+    type: z.literal("one_time_notice"),
+});
+
+const onboarding_steps_schema = z.union([one_time_notice_schema, hotspot_schema]);
+
+export type OnboardingSteps = z.output<typeof onboarding_steps_schema>;
```

Comment:
This is the type of one onboarding step, not the type of a collection of steps. It should have been named `onboarding_step_schema` and `OnboardingStep`, not the plural.

Reply:
Can I change it in the PR of hotspots.ts migration ?